code/modules/mob/inventory.dm

```diff
@@ -128,15 +128,25 @@
 	Arguments
 	* obj/item/W is the item you are trying to equip
 	* del_on_fail if true will delete the item instead of dropping it to the floor
+	* merge_stacks if true will merge stacks that src is holding in its hands with new ones
 
 	Returns TRUE if it was able to put the thing into one of our hands.
 */
-/mob/proc/put_in_hands(obj/item/W, del_on_fail = FALSE)
+/mob/proc/put_in_hands(obj/item/W, del_on_fail = FALSE, merge_stacks = TRUE)
 	W.do_pickup_animation(src)
 	if(status_flags & INCORPOREAL) // INCORPOREAL things don't have hands
 		return FALSE
 	if(!W)
 		return FALSE
+	if(merge_stacks && istype(W, /obj/item/stack))
+		var/obj/item/stack/new_stack = W
+		var/obj/item/stack/stack_in_hands = null
+		if(istype(l_hand, new_stack.merge_type))
+			stack_in_hands = l_hand
+		else if(istype(r_hand, new_stack.merge_type))
+			stack_in_hands = r_hand
+		if(stack_in_hands && new_stack.merge(stack_in_hands))
+			return TRUE
```

Comment:
this is very smelly. This should be an item proc, not a mob proc.

Reply:
`user.put_in_hands(stack)`, one could argue it should be a mob proc because it involves mob and mob hands. and what if mob does not have hands? then you check type of a mob like you check type of a stack here.
I don't want to make it super complicated and change a bunch of code just for such simple thing as merging stacks.
KISS